internal/archive_status_manager_fake.go

```diff
@@ -0,0 +1,42 @@
+package internal
```

Comment:
Oh wait, I just noticed that I can do `NewDataFolderASM(&MockDataFolder)` instead of this `FakeASM`. I'll look into implementing that instead (possibly just by converting fake to use that as an implementation) - "MockDataFolder-backed DataFolderASM"

Reply:
Upon second thought, this seems lightweight enough that its harmless to leave as-is. Happy to implement if someone feels strongly.